api/run.py

```diff
@@ -439,8 +462,8 @@ def getStudentProfile():
     for parentId in parentIds:
         parent = parentsDOM.getParentProfile(parentId)
         studentsOfParent = []
-        for studentID in parent['student_ids']:
-            student = studentsDOM.getFullInfo(studentID)
+        for currStudentID in parent['student_ids']:
```

Comment:
@vietnguyen00 this had actually caused a problem with variable name shadowing below... `studentID` was the same name as the `studentID` passed in the student profile, and when the later `studentsDOM.getBasicInfo()` was called, it used a the last `studentID` that came from this `for` loop, instead of the original `studentID` that was in the URL parameters! Fixed now.